src/test/groovy/org/prebid/server/functional/tests/module/AnalyticsTagsModuleSpec.groovy

```diff
@@ -0,0 +1,274 @@
+        def account = new Account(uuid: bidRequest.accountId, config: accountConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        def bidResponse = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should not contain any analytics tag"
+        assert !bidResponse.ext.prebid.analytics.tags
+
+        and: "Bid response shouldn't contain warning"
+        assert !bidResponse.ext.warnings
+    }
+
+    def "PBS should not include analytics tag in response without any warnings when client details is disabled in request"() {
+        given: "Default account with module config"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            it.ext.prebid.analytics = new PrebidAnalytics(options: new AnalyticsOptions(enableClientDetails: false))
+        }
+
+        and: "Account in the DB"
+        def executionPlan = ExecutionPlan.getSingleEndpointExecutionPlan(OPENRTB2_AUCTION, ORTB2_BLOCKING, [RAW_BIDDER_RESPONSE])
+        def hooksConfiguration = new AccountHooksConfiguration(executionPlan: executionPlan)
+        def accountConfig = new AccountConfig(hooks: hooksConfiguration, analytics: new AccountAnalyticsConfig(allowClientDetails: true))
+        def account = new Account(uuid: bidRequest.accountId, config: accountConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        def bidResponse = pbsServiceWithEnabledOrtb2Blocking.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should not contain any  analytics tag"
+        assert !bidResponse.ext.prebid.analytics
+
+        and: "Bid response shouldn't contain warning"
+        assert !bidResponse.ext.warnings
+    }
+
+    def "PBS should not include analytics tag in response with warning when client details is disabled in account"() {
+        given: "Default account with module config"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            it.ext.prebid.analytics = new PrebidAnalytics(options: new AnalyticsOptions(enableClientDetails: true))
+        }
+
+        and: "Account in the DB"
+        def executionPlan = ExecutionPlan.getSingleEndpointExecutionPlan(OPENRTB2_AUCTION, ORTB2_BLOCKING, [RAW_BIDDER_RESPONSE])
+        def hooksConfiguration = new AccountHooksConfiguration(executionPlan: executionPlan)
+        def accountConfig = new AccountConfig(hooks: hooksConfiguration, analytics: new AccountAnalyticsConfig(allowClientDetails: false))
+        def account = new Account(uuid: bidRequest.accountId, config: accountConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        def bidResponse = pbsServiceWithEnabledOrtb2Blocking.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should not contain any analytics tag"
+        assert !bidResponse.ext.prebid.analytics
```

Comment:
My propose to replace on `!bidResponse?.ext?.prebid?.analytics?.tags` for better readability.
Check other occurrences.